internal/test/framework/framework.go

```diff
@@ -369,6 +395,100 @@ func (f *TestFramework) RetrieveArtifacts() {
 	}
 }
 
+// ApplyHybridOverlayPatch will enable the hybrid overlay on the cluster
+func (f *TestFramework) ApplyHybridOverlayPatch() error {
+	jsonPatch := []byte(`{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"hybridOverlayConfig":` +
+		`{"hybridClusterNetwork":[{"cidr":"10.132.0.0/14","hostPrefix":23}]}}}}}`)
+
+	_, err := f.OSOperatorClient.Networks().Patch("cluster", k8stypes.MergePatchType, jsonPatch)
+	if err != nil {
+		return fmt.Errorf("could not apply patch %s to cluster network: %s", string(jsonPatch), err)
+	}
+
+	// Wait until the hybrid overlay changes are complete
+	return f.waitUntilHybridOverlayReady()
+}
+
+// waitUntilHybridOverlayReady returns once OVN is ready again, after applying the hybrid overlay patch.
+func (f *TestFramework) waitUntilHybridOverlayReady() error {
+	err := f.waitUntilMasterNodesAnnotated()
+	if err != nil {
+		return err
+	}
+
+	// This is being done after we wait for the master nodes, as we need to make sure the patch is being acted on,
+	// and we are not checking the pods before they begin to restart with the hybrid overlay changes
+	err = f.waitUntilOVNPodsReady()
```

Comment:
nit: You can use `if err:=f.waitUntil..` pattern here.

Reply:
I would say this is not a nit and is a pattern we should follow where possible.